File: com.systemincloud.examples.tasks.javatask/src/test/java/tasks/generator/GeneratorAtomicWith1SIn.java
package tasks.generator;

import com.systemincloud.modeler.tasks.javatask.api.InputPort;
import com.systemincloud.modeler.tasks.javatask.api.JavaTask;
import com.systemincloud.modeler.tasks.javatask.api.OutputPort;
import com.systemincloud.modeler.tasks.javatask.api.annotations.InputPortInfo;
import com.systemincloud.modeler.tasks.javatask.api.annotations.JavaTaskInfo;
import com.systemincloud.modeler.tasks.javatask.api.annotations.OutputPortInfo;
import com.systemincloud.modeler.tasks.javatask.api.data.Int32;
import com.systemincloud.modeler.tasks.javatask.api.data.Text;

@JavaTaskInfo(generator = true)
public class GeneratorAtomicWith1SIn extends JavaTask {

	@InputPortInfo(name = "In", dataType = Int32.class)
	public InputPort in;
	@OutputPortInfo(name = "Out", dataType = Text.class)
	public OutputPort out;

	@Override
	public void generate() {
		out.putData(new Text("Hello World"));
		sleep(1000);
	}

	@Override
	public void execute(int grp) {
		out.putData(new Text(Integer.toString(in.getData(Int32.class).getValue())));
	}

}

File: com.systemincloud.examples.tasks.pythontask/src/test/py/tasks/generator/GeneratorAtomicWith1SIn.py
from sicpythontask.PythonTaskInfo import PythonTaskInfo
from sicpythontask.PythonTask import PythonTask
from sicpythontask.InputPort import InputPort
from sicpythontask.OutputPort import OutputPort
from sicpythontask.data.Int32 import Int32
from sicpythontask.data.Text import Text

@PythonTaskInfo(generator=True)
class GeneratorAtomicWith1SIn(PythonTask):

    def __init_ports__(self):
        self.in_ = InputPort(name="in", data_type=Int32)
        self.out = OutputPort(name="out", data_type=[Int32, Text])

    def generate(self):
        self.out.put_data(Text("Hello World"));
        self.sleep(1000);

    def execute(self, grp):
        inValue = self.in_.get_data(Int32).values[0]
        self.out.put_data(Int32(2*inValue))

